avocado/core/multiplexer.py

```diff
@@ -377,28 +377,67 @@ def iteritems(self):
                 yield (leaf.environment_origin[key].path, key, value)
 
 
+def _report_already_parsed(self, *args, **kwargs):
+    """
+    Raises exception describing that `self.data` alteration is restricted
+    """
+    raise RuntimeError("Altering of `self.data` is restricted. %s %s"
```

Comment:
BTW, any reason for defining this outside the scope of the class? That is, any reason not to do this:

```diff
diff --git a/avocado/core/multiplexer.py b/avocado/core/multiplexer.py
index 224d2c8..a32b0b9 100644
--- a/avocado/core/multiplexer.py
+++ b/avocado/core/multiplexer.py
@@ -363,14 +363,6 @@ class AvocadoParam(object):
                 yield (leaf.environment_origin[key].path, key, value)
 
 
-def _report_already_parsed(self, *args, **kwargs):
-    """
-    Raises exception describing that `self.data` alteration is restricted
-    """
-    raise RuntimeError("Altering of `self.data` is restricted. %s %s"
-                       % (args, kwargs))
-
-
 class Mux(object):
 
     """
@@ -383,6 +375,14 @@ class Mux(object):
         self.data = tree.TreeNodeDebug() if debug else tree.TreeNode()
         self._mux_path = None
 
+    @staticmethod
+    def _report_already_parsed(*args, **kwargs):
+        """
+        Raises exception describing that `self.data` alteration is restricted
+        """
+        raise RuntimeError("Altering of `self.data` is restricted. %s %s"
+                           % (args, kwargs))
+
     def parse(self, args):
         """
         Apply options defined on the cmdline
@@ -396,8 +396,8 @@ class Mux(object):
             self._mux_path = ['/run/*']
         # disable data alteration (and remove data as they are not useful)
         self.data = None
-        self.data_inject = _report_already_parsed
-        self.data_merge = _report_already_parsed
+        self.data_inject = self._report_already_parsed
+        self.data_merge = self._report_already_parsed
 
     @property
     def is_parsed(self):

```